lib/Optimizer/CodeGen/LowerToBaseProfileQIR.cpp

```diff
@@ -556,3 +566,43 @@ void cudaq::opt::registerBaseProfilePipeline() {
       "Pass pipeline to generate code for the QIR base profile.",
       addBaseProfilePipeline);
 }
+
+namespace cudaq {
+/// Remove Measurements
+///
+/// This pass removes measurements and the corresponding output recording calls.
+/// This is needed for backends that don't support selective measurement calls.
+/// For example: https://github.com/NVIDIA/cuda-quantum/issues/512
+struct RemoveMeasurementsPass
+    : public cudaq::opt::QIRToBaseQIRBase<RemoveMeasurementsPass> {
```

Comment:
Instead of creating a pass that inherits another full-fledged pass, maybe we can try adding a pass option to the QIRToBaseQIR pass to control this. The option would, I imagine, just add a couple/few different rewrite patterns vs. the pass as it stands.

Reply:
Per our discussion, I think this does indeed need to be a new pass because we need the "regular" QIR passes to run before this change. However, I updated the code to use a new dedicated pass rather than hijacking `QIRToBaseQIRBase` like I did above. I believe that should address your concern. Updated in [0893654](https://github.com/NVIDIA/cuda-quantum/pull/661/commits/08936549ab2f5342da263dca160681200b209e55).